framework/elements/BaseElement.js

```diff
@@ -0,0 +1,169 @@
+        } catch (error) {
+            isExisting = false;
+        }
+        Logger.debug(`The result of isExisting for ${this.name} is ${isExisting}`)
+        return isExisting;
+    }
+
+    async isEnabled() {
+        Logger.debug(`Checking if ${this.name} is enabled`)
+        const element = await this.getElement(this.selector);
+        let isEnabled;
+        try {
+            isEnabled = await element.isEnabled();
+        } catch (error) {
+            isEnabled = false;
+        }
+        Logger.debug(`The result of isEnabled for ${this.name} is ${isEnabled}`)
+        return isEnabled;
+    }
+
+    async isSelected() {
+        Logger.debug(`Checking if ${this.name} is selected`)
+        const element = await this.getElement(this.selector);
+        let isSelected;
+        try {
+            isSelected = await element.isSelected();
+        } catch (error) {
+            isSelected = false;
+        }
+        Logger.debug(`The result of isSelected for ${this.name} is ${isSelected}`)
+        return isSelected;
+    }
+
+    async getCssProperty(propertyName) {
+        Logger.debug(`Getting CSS property ${propertyName} of ${this.name}`)
+        const element = await this.getElement(this.selector);
+        const cssProperty = await element.getCSSProperty(propertyName);
+        Logger.debug(`The CSS property ${propertyName} of ${element} is ${cssProperty}`)
+        return cssProperty
+    }
+
+    async scrollIntoView() {
+        Logger.debug(`Scroll ${this.name} into view`)
+        const element = await this.getElement(this.selector);
+        element.scrollIntoView();
+    }
+
+    async getText() {
+        Logger.debug(`Get the text of element ${this.name}`)
+        const element = await this.getElement();
+        const elementText = await element.getText();
+        Logger.debug(`The text of ${element} is ${elementText}`)
+        return elementText;
+    }
+
+    async dragAndDrop(targetSelector) {
+        Logger.debug(`Moving ${this.name} to ${targetSelector}`)
+        const sourceElement = await this.getElement(this.selector);
+        const targetElement = await browser.$(targetSelector);
+        sourceElement.dragAndDrop(targetElement);
+    }
+
+    async click() {
+        Logger.debug(`Clicking ${this.name}`)
+        const element = await this.getElement(this.selector);
+        element.click();
+    }
+
+    async doubleClick() {
+        Logger.debug(`Double click ${this.name}`)
+        const element = await this.getElement(this.selector);
+        element.doubleClick();
+    }
+
+    async rightClick() {
+        Logger.debug(`Right click ${this.name}`)
+        const element = await this.getElement(this.selector);
+        element.click({ button: 'right' });
+    }
+
+    async waitForExist(selector, DEFAULT_TIMEOUT) {
+        Logger.debug(`Waiting for ${this.name} to exist`)
+        const element = await this.getElement(selector);
+        element.waitForExist({ DEFAULT_TIMEOUT });
```

Comment:
such methods throw error in case if condition not met. please wrap them with try catch